src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -178,24 +179,30 @@ private static Glyph GetGlyph(SymbolKind kind, Accessibility? declaredAccessibil
 
             if (type is INamedTypeSymbol namedType && namedType.OriginalDefinition != null)
             {
-                var originalDefinition = namedType.OriginalDefinition;
-
-                var ienumerableOfT = namedType.GetAllInterfacesIncludingThis().FirstOrDefault(
-                    t => t.OriginalDefinition.SpecialType == SpecialType.System_Collections_Generic_IEnumerable_T);
+                var getEnumeratorMethod = namedType.GetMembers()
+                    .OfType<IMethodSymbol>()
+                    .FirstOrDefault(m => m.IsValidGetEnumerator() || m.IsValidGetAsyncEnumerator());
```

Comment:
So system.string implements this interface. Can you add a test that shows this won't apply to that type when naming? (And, if it does, blacklist that case).

Reply:
@CyrusNajmabadi This is already tested in `NoSuggestionsForString`:

https://github.com/dotnet/roslyn/blob/3655d5490eef0df4717087caaaeb2ca85fed1eca/src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs#L651-L661

The part that skips it is the last line of the following method:

https://github.com/dotnet/roslyn/blob/9f82a08241f46854e2a1ccb0c3f40195a2c7a70b/src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs#L109-#L127